File: java/src/model/Personaje.java
package model;

// Clase padre
public class Personaje {

    // Encapsulamiento:
    // los atributos son private
    private int id;
    private String nombre;
    private int nivel;

    // Constructor
    public Personaje(int id, String nombre, int nivel) {
        this.id = id;
        this.nombre = nombre;
        this.nivel = nivel;
    }

    // Getter del id
    public int getId() {
        return id;
    }

    // Setter del id
    public void setId(int id) {
        this.id = id;
    }

    // Getter del nombre
    public String getNombre() {
        return nombre;
    }

    // Setter del nombre
    public void setNombre(String nombre) {

        // Validación básica
        if(nombre.length() > 0){
            this.nombre = nombre;
        }
    }

    // Getter del nivel
    public int getNivel() {
        return nivel;
    }

    // Setter del nivel
    public void setNivel(int nivel) {

        // Evita niveles negativos
        if(nivel > 0){
            this.nivel = nivel;
        }
    }

    // Método polimórfico
    // las clases hijas lo sobrescriben
    public void habilidadEspecial(){
        System.out.println("Habilidad básica");
    }
}

File: python/model/personaje.py
class Personaje:

    # Constructor
    def __init__(self, id, nombre, nivel):

        # atributos privados
        self.__id = id
        self.__nombre = nombre
        self.__nivel = nivel

    # Getter id
    @property
    def id(self):
        return self.__id

    # Setter id
    @id.setter
    def id(self, id):
        self.__id = id

    # Getter nombre
    @property
    def nombre(self):
        return self.__nombre

    # Setter nombre
    @nombre.setter
    def nombre(self, nombre):

        # validación
        if len(nombre) > 0:
            self.__nombre = nombre

    # Getter nivel
    @property
    def nivel(self):
        return self.__nivel

    # Setter nivel
    @nivel.setter
    def nivel(self, nivel):

        if nivel > 0:
            self.__nivel = nivel

    # Método polimórfico
    def habilidad_especial(self):

        print("Habilidad básica")